src/openrct2/management/Research.cpp

```diff
@@ -540,8 +540,12 @@ bool research_insert_ride_entry(uint8_t rideType, ObjectEntryIndex entryIndex, u
 {
     if (rideType != RIDE_TYPE_NULL)
     {
-        research_insert(
-            { static_cast<uint32_t>(RESEARCH_ENTRY_RIDE_MASK | (rideType << 8) | entryIndex), category }, researched);
+        ResearchItem tmpItem = {};
```

Comment:
Why create a temporary then initialise it. Can't you create a constructor in research item for this.